docs/current/guides/488564-openshift-gitlab.md

```diff
@@ -0,0 +1,111 @@
+  ```
+
+  This configuration uses the label `builder-node=true` to taint the nodes on which the Dagger Engine should be deployed.
+
+1. Execute the following command for each node that is intended to host a Dagger Engine (replace the `NODE-NAME` placeholder with each node name):
+
+  ```shell
+  oc adm taint nodes NODE-NAME builder-node=true:NoSchedule
+  ```
+
+1. Install the Dagger Engine using the Dagger Helm chart:
+
+  ```shell
+  helm upgrade --create-namespace --install --namespace dagger dagger oci://registry.dagger.io/dagger-helm -f values.yaml
+  ```
+
+1. Grant the necessary permissions for the `default` service account in the `dagger` namespace:
+
+  :::info
+  Without this step, pod creation will fail due to insufficient permissions to execute privileged containers with fixed user IDs and host path volume mounts.
+  :::
+
+  ```shell
+  oc adm policy add-scc-to-user privileged -z default -n dagger
+  ```
+
+## Step 2: Configure GitLab Runner
+
+The next step is to configure a GitLab Runner. Follow these steps:
+
+1. Create the configuration for the GitLab Runner as `runner-config.yaml` and `runner.yaml`. Replace the `YOUR-GITLAB-URL` placeholder with the URL of your GitLab instance.
+
+  ```yaml title=runner-config.yaml file=./snippets/openshift-gitlab/runner-config.yaml
+  ```
+
+  ```yaml title=runner.yaml file=./snippets/openshift-gitlab/runner.yaml
+  ```
+For the `gitlabUrl`, use the URL for your GitLab instance. For example, if your project is hosted on gitlab.example.com/yourname/yourproject, your GitLab instance URL is https://gitlab.example.com.
+For the `spec.token` follow the instructions on [How create a runner](https://docs.gitlab.com/ee/ci/runners/runners_scope.html#create-a-shared-runner-with-a-runner-authentication-token) to obtain a token.
+
+  This configuration uses a similar configuration as that seen in Step 1 for the taints and tolerations and the pod affinity. This ensures that the GitLab runner worker pods only runs on nodes with Dagger engines.
+
+1. Apply the configuration and deploy the GitLab Runner:
+
+  ```shell
+  oc apply -f runner-config.yaml -n dagger
+  oc apply -f runner.yaml -n dagger
+  ```
+
+## Step 3: Create a GitLab CI/CD pipeline
+
+Create a new GitLab CI/CD pipeline configuration file in your repository at `.gitlab-ci.yml` with the following content:
+
+```yaml title=.gitlab-ci.yml file=./snippets/openshift-gitlab/.gitlab-ci.yml
+```
+
+The most important section of this configuration are:
+
+- The `tags` entry, which tells GitLab to use the GitLab Runner which is connected to the Dagger Engine; and
+- The `_EXPERIMENTAL_DAGGER_RUNNER_HOST` variable, which specifies the socket for the Dagger CLI to use.
+
+To connect your GitLab runners with Dagger Cloud, [add a new CI/CD variable in GitLab](https://docs.gitlab.com/ee/ci/variables/#define-a-cicd-variable-in-the-ui) with the name `DAGGER_CLOUD_TOKEN` and set as value the token from the organization in the Dagger Cloud.
```

Comment:
Don't we need to also reference the secret in the `.gitlab-ci.yml` file? See similar question in that file.

Reply:
CI/CD Variables are injected automatically.
So every Job already knows the `DAGGER_CLOUD_TOKEN` variable.
It would only necessary if you remap the name, for example
`"DAGGER_CLOUD_TOKEN": $DAGGER_CLOUD_SECRET_TOKEN`